sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1544,8 +1544,17 @@ class AstBuilder extends SqlBaseParserBaseVisitor[AnyRef] with SQLConfHelper wit
         if (name.length > 1) {
           throw QueryParsingErrors.invalidTableValuedFunctionNameError(name, ctx)
         }
+        val args = func.functionArgument.asScala.map { e =>
+          if (e.namedArgumentExpression != null) {
+            val key = e.namedArgumentExpression.key.getText
+            val value = e.namedArgumentExpression.value
+            NamedArgumentExpression(key, expression(value))
```

Comment:
Can you control this behavior with a SQLConf? You can create a new entry like `ALLOW_NAMED_FUNCTION_ARGUMENTS` and return an error here if the config is not set to true:

https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala